game-engine/domain/src/main/java/za/co/entelect/challenge/entities/TowerDefenseGameMap.java

```diff
@@ -118,14 +120,29 @@ public void removeMissile(Missile missile) {
         this.missiles.remove(missile);
     }
 
-    public void moveMissileSingleSpace(Missile p) throws Exception {
+    public void moveMissileSingleSpace(Missile p) {
         int newPosition = p.getX() + (p.getDirection().getMultiplier());
 
         boolean homeBaseIsHit = (newPosition < 0 || newPosition >= GameConfig.getMapWidth());
         if (homeBaseIsHit) {
-            TowerDefensePlayer opponent = getPlayerOpponent(p.playerType);
-            TowerDefensePlayer missileOwner = getPlayer(p.playerType);
-            opponent.takesHitByPlayer(p, missileOwner);
+
+            TowerDefensePlayer opponent = null;
+            try {
+                opponent = getPlayerOpponent(p.playerType);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            TowerDefensePlayer missileOwner = null;
+            try {
+                missileOwner = getPlayer(p.playerType);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
```

Comment:
Not sure if this is a bad suggestion, but can we not merge these two try catch statements. Seems pretty redundant, but then again it probably was done.

I can see that you might want missleOwner to still initialize regardless of whether opponent gets initialized, but looking at the rest of the method, if opponent is broken, missleOwner is pointless.

Reply:
Yes that is correct. My plan is to get very descriptive exception messages in the log files when the game-engine messes up. I was thinking with these we can have distinct message depending on what happened

Reply:
Makes sense 👍